src/main.c

```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
 		return 1;
 	}
 
-	int result = 0, pos, player = 0;
+	int result = 0, pos, player = 0,i,sw=0;
 	const char Tic[] = {'X', 'O'};
-	char Board[N * N];
+	char Board[N * N],c;
+	char* s;
 
 	init(N, Board);
 	greet();
 
 	while (show(N, Board), !result) {
 		do {
 			printf("\nFor Player(\'%c\'):\n", Tic[player]);
-			printf("Enter Position: "); 
-			assert(scanf("%d", &pos) == 1);
+			printf("Enter Position (non-digit characters not allowed.. not even white spaces...): ");
+			goto init__1;
```

Comment:
We're strictly against of use of `goto`. Try alternatives if possible. @sammy00747